src/senselab/audio/tasks/speaker_verification/speaker_verification.py

```diff
@@ -0,0 +1,59 @@
+"""Audio Processing and Speaker Verification Module.
+
+This module provides functions for resampling audio using an IIR filter and
+verifying if two audio samples or files are from the same speaker using a
+specified model.
+"""
+
+from typing import List, Tuple
+
+from torch.nn.functional import cosine_similarity
+
+from senselab.audio.data_structures.audio import Audio
+from senselab.audio.tasks.speaker_embeddings.speechbrain import SpeechBrainEmbeddings
+from senselab.utils.data_structures.device import DeviceType
+from senselab.utils.data_structures.model import SpeechBrainModel
+
+
+def verify_speaker(
+    audios: List[Tuple[Audio, Audio]],
+    model: SpeechBrainModel = SpeechBrainModel(path_or_uri="speechbrain/spkrec-ecapa-voxceleb", revision="main"),
+    model_training_sample_rate: int = 16000,  # spkrec-ecapa-voxceleb trained on 16kHz audio
```

Comment:
this is not a user parameter, but a configuration variable you should get from the model configuration. please, see https://github.com/sensein/senselab/blob/main/src/senselab/audio/tasks/speech_enhancement/speechbrain.py

Reply:
I'm pretty sure the training sample rate is not available from the model configuration for `speechbrain/spkrec-ecapa-voxceleb`.

I checked the output of the instance methods of `model = SpeechBrainModel(path_or_uri="speechbrain/spkrec-ecapa-voxceleb", revision="main")` (e.g. `model.get_model_info()`).

I also get an error when I run:
```
def get_model_sample_rate(model, device=DeviceType.CPU):
    enhancer = SpeechBrainEnhancer._get_speechbrain_model(model=model, device=device)
    return enhancer.hparams.sample_rate

# Usage example
model = SpeechBrainModel(path_or_uri="speechbrain/spkrec-ecapa-voxceleb", revision="main")
sample_rate = get_model_sample_rate(model)
print(f"The model's sample rate is: {sample_rate} Hz")
```
Error output:
```
'types.SimpleNamespace' object has no attribute 'sample_rate'
```

However, the above code works when I run it with `speechbrain/sepformer-wham16k-enhancement`.

Reply:
you are so right. in the other use case when we use ecapa-tdnn we fixed `expected_sampling_rate` to 16khz in the code (https://github.com/sensein/senselab/blob/main/src/senselab/audio/tasks/speaker_embeddings/speechbrain.py), since all speechbrain models for speaker embeddings work at 16khz. it's not ideal, but a good workaround. I think you can hardcode 16khz in the code, too and remove it from the params. we want to remove any chance for the user to make silly mistakes. thanks!! and nice catch!!

Reply:
Thanks! Done.